files/scripts/syncd.sh

```diff
@@ -0,0 +1,142 @@
+#!/bin/bash
+
+SERVICE="syncd"
+PEER="swss"
+DEBUGLOG="/tmp/swss-syncd-debug.log"
+
+function debug()
+{
+    /bin/echo `date` "- $1" >> ${DEBUGLOG}
+}
+
+function lock_service_state_change()
+{
+    debug "Locking ${SERVICE} service"
+    svclock=$(/bin/mktemp /tmp/${SERVICE}-service-lock.XXXXXX)
+    trap "/bin/rm -f $svclock" 0 2 3 15
+    peerlock=`/bin/ls /tmp/${PEER}-service-lock* 2>/dev/null`
+    [[ -n "$peerlock" ]] && debug "Found ${PEER} service lock: $peerlock"
+}
+
+function unlock_service_state_change()
+{
+    debug "Unlocking ${SERVICE} service"
+    /bin/rm -f $svclock
+}
+
+function check_warm_boot()
+{
+    SYSTEM_WARM_START=`/usr/bin/redis-cli -n 4 hget "WARM_RESTART|system" enable`
+    SERVICE_WARM_START=`/usr/bin/redis-cli -n 4 hget "WARM_RESTART|${SERVICE}" enable`
+    if [[ x"$SYSTEM_WARM_START" == x"true" ]] || [[ x"$SERVICE_WARM_START" == x"true" ]]; then
+        WARM_BOOT="true"
+    else
+        WARM_BOOT="false"
+    fi
+}
+
+function validate_restart_count()
+{
+    if [[ x"$WARM_BOOT" == x"true" ]]; then
+        RESTART_COUNT=`/usr/bin/redis-cli -n 6 hget "WARM_RESTART_TABLE|orchagent" restart_count`
```

Comment:
Would it make more sense to check syncd restart_count?   
Also in https://github.com/Azure/sonic-swss/pull/600,  @rodnymolina is to change it to restore_count,   any comment there?

Reply:
As Jipan and I discussed offline. For now, I removed code to support syncd service warm boot.

Once this feature gets properly designed and ready for coding/testing, we can put these code back.